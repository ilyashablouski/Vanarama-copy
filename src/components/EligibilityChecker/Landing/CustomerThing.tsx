import dynamic from 'next/dynamic';
import React, { FC } from 'react';
import Skeleton from '../../Skeleton';

const Text = dynamic(() => import('core/atoms/text'), {
  loading: () => <Skeleton count={1} />,
});
const Heading = dynamic(() => import('core/atoms/heading'), {
  loading: () => <Skeleton count={1} />,
});
const EligibilityCheckerButton = dynamic(
  () => import('./EligibilityCheckerButton'),
  {
    loading: () => <Skeleton count={1} />,
  },
);

interface ICustomerThing {
  heading: string | null;
  description: string | null;
}

const CustomerThing: FC<ICustomerThing> = ({ heading, description }) => (
  <div className="row:lead-text">
    <Heading size="large" color="black">
      {heading}
    </Heading>
    <Text tag="p" size="regular" color="darker">
      {description}
    </Text>
    <EligibilityCheckerButton />
  </div>
);

export default CustomerThing;
