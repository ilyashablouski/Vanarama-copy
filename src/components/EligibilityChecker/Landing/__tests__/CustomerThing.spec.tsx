import React from 'react';
import renderer from 'react-test-renderer';
import CustomerThing from '../CustomerThing';

const getComponent = () => {
  return renderer.create(<CustomerThing />).toJSON();
};

describe('<CustomerThing />', () => {
  it('renders correctly', () => {
    const tree = getComponent();
    expect(tree).toMatchSnapshot();
  });
});
