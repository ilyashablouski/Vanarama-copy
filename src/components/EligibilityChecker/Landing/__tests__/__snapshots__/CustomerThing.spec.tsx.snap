// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomerThing /> renders correctly 1`] = `
<section
  className="section"
>
  <div
    className="container"
  >
    <div>
      <div
        className="-col-400 -a-center"
      >
        <p
          className="heading -black -large"
        >
          What Do Our Customers Think?
        </p>
        <p
          className="text -expand -darker"
        >
          More than 1,000 people use our Eligibility Checker every week to find out their likelihood of getting accepted to lease a new car lease.
        </p>
        <button
          className="button -teal -regular -solid"
          data-testid="eligibility-Checker-btn"
          onClick={[Function]}
        >
          <div
            className="button--inner"
          >
            Check Your Eligibility
          </div>
        </button>
      </div>
    </div>
  </div>
</section>
`;
